edb/edgeql/compiler/inference/cardinality.py

```diff
@@ -355,7 +355,7 @@ def __infer_set(
             else:
                 rptrref_card = rptrref.dir_cardinality
 
-            if rptrref_card.is_single():
+            if rptrref_card is not None and rptrref_card.is_single():
```

Comment:
This is suspicious.  In what scenario is it `None`?

Reply:
Try the following in `graphqlfunctional` test database:
```
SELECT default::SettingAliasAugmented {
    of_group: {
        name_upper
    }
} FILTER .of_group.name_upper > "U";
```
(This is also triggered by the `test_graphql_functional_alias_05` from this PR)

I was using the reasoning that in an "if/else" situation everything that's not explicitly in the "if" is intended to fall under else, so I didn't think too much before adding the `is not None` part. You're right that it may indicate a different bug instead.

Reply:
This was resolved by a separate issue.